common/Resources/ti.internal/extensions/node/buffer.js

```diff
@@ -61,1446 +53,1543 @@ const uint8DoubleArray = new Uint8Array(doubleArray.buffer);
-		Object.defineProperties(this, {
-			byteOffset: {
-				value: start
-			},
-			length: {
-				value: length
-			},
-			_tiBuffer: {
-				value: tiBuffer
-			}
-		});
+/**
+ * Constructs a new buffer.
+ *
+ * Primarily used internally in this module together with `newBuffer` to
+ * create a new Buffer instance wrapping a Ti.Buffer.
+ *
+ * Also supports the deprecated Buffer() constructors which are safe
+ * to use outside of this module.
+ *
+ * @param {integer[]|Buffer|integer|string|Ti.Buffer} arg the underlying data/bytes
+ * @param {string|integer} encodingOrOffset encoding of the string, or start offset of array/buffer
+ * @param {integer} length length of the underlying array/buffer to wrap
+ * @returns {Buffer}
+ */
+function Buffer(arg, encodingOrOffset, length) {
+	if (arg === undefined) {
+		return;
 	}
 
-	// This is a method we should get by extending Uint8Array, so really should only be overriden on a "SlowBuffer" that wraps Ti.Buffer
-	get buffer() {
-		// Get the slice of the array from byteOffset to length
-		return Uint8Array.from(this).buffer;
-	}
+	if (typeof arg !== 'object' || arg.apiName !== 'Ti.Buffer') {
+		showFlaggedDeprecation();
 
-	// This is a method we should get by extending Uint8Array, so really should only be overriden on a "SlowBuffer" that wraps Ti.Buffer
-	set(src, offset = 0) {
-		const numBytes = src.length;
-		// check src.length + offset doesn't go beyond our length!
-		checkOffset(this, offset, numBytes);
-		// copy src values into this buffer starting at offset
-		for (let i = 0; i < numBytes; i++) {
-			setAdjustedIndex(this, i + offset, src[i]);
+		if (typeof arg === 'number') {
+			if (typeof encodingOrOffset === 'string') {
+				throw new TypeError(`The "string" argument must be of type "string". Received type ${typeof arg}`);
+			}
+			return Buffer.alloc(arg);
 		}
+		return Buffer.from(arg, encodingOrOffset, length);
 	}
 
-	/**
-	 * 0 is returned if target is the same as buf
-	 * 1 is returned if target should come before buf when sorted.
-	 * -1 is returned if target should come after buf when sorted.
-	 * @param {Buffer} target Buffer to compare against
-	 * @param {integer} [targetStart=0] index to start in target
-	 * @param {integer} [targetEnd=target.length] index to end in target
-	 * @param {integer} [sourceStart=0] index to start in this Buffer
-	 * @param {integer} [sourceEnd=this.length] index to end in this Buffer
-	 * @returns {integer}
-	 */
-	compare(target, targetStart, targetEnd, sourceStart, sourceEnd) {
-		if (!Buffer.isBuffer(target)) {
-			throw new TypeError(`The "target" argument must be one of type Buffer or Uint8Array. Received type ${typeof buf1}`);
-		}
-		if (targetStart === undefined) {
-			targetStart = 0;
+	// The slow case - we're wrapping a Ti.Buffer
+	return SlowBuffer.fromTiBuffer(arg, encodingOrOffset, length);
+}
+
+/**
+ * @param {integer[]|Buffer|string} value value we're wrapping
+ * @param {string|number} [encodingOrOffset]
+ * @param {number} [length]
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/buffer.js;
  Line: 103;
-->

- :warning: common/Resources/ti.internal/extensions/node/buffer.js line 103 – Missing JSDoc parameter description for 'length'. (valid-jsdoc)